pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GapfillProcessor.java

```diff
@@ -270,7 +275,7 @@ private List<Object[]> aggregateGapfilledData(Object timeCol, List<Object[]> buc
     }
 
     Map<ExpressionContext, BlockValSet> blockValSetMap = new HashMap<>();
-    for (int i = 1; i < dataSchema.getColumnNames().length; i++) {
+    for (int i = 0; i < dataSchema.getColumnNames().length; i++) {
```

Comment:
Can you explain this change? Previously we were skipping block val set for index 0, but now we track all the column in that map. Does it lead to any behavior change?